File: sherpout-server/src/main/java/com/sherpout/server/api/exercise/enumerated/ExerciseEquipment.java
package com.sherpout.server.api.exercise.enumerated;

import com.sherpout.server.commons.enumerated.BaseEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum ExerciseEquipment implements BaseEnum {
    BARBELL(1),
    DUMBBELL(2),
    KETTLEBELL(3),
    RESISTANCE_BAND(4),
    MACHINE(5),
    CABLE_MACHINE(6),
    PLATFORM(7),
    BENCH(8),
    PULL_UP_BAR(9);

    private final Integer id;
}

File: sherpout_mobile/lib/common/dto/exercise_equipment.dart
import 'package:flutter/cupertino.dart';

enum ExerciseEquipment {
  barbell("barbell", "sztanga"),
  dumbell("dumbell", "hantel"),
  kettlebell("kettlebell", "odważnik"),
  resistanceBand("resistance band", "guma oporowa"),
  machine("machine", "maszyna"),
  cableMachine("cable machine", "wyciąg"),
  platform("platform", "platforma"),
  bench("bench", "ławka"),
  pullUpBar("pull-up bar", "drążek");

  final String en;
  final String pl;

  const ExerciseEquipment(this.en, this.pl);

  static ExerciseEquipment? fromName(String name) {
    return ExerciseEquipment.values.firstWhere(
          (e) => e.name.toUpperCase() == name.replaceAll("_", ""),
    );
  }

  String localized(BuildContext context) {
    final lang = Localizations.localeOf(context).languageCode;
    switch (lang) {
      case 'pl':
        return pl;
      case 'en':
      default:
        return en;
    }
  }
}